src/environment_cache.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+)
+
+const (
+	cachePath = "/.omp.cache"
+)
+
+type fileCache struct {
+	cache *concurrentMap
+	home  string
+}
+
+func (fc *fileCache) init(home string) {
+	fc.cache = newConcurrentMap()
+	fc.home = home
+	content, err := ioutil.ReadFile(fc.home + cachePath)
+	if err != nil {
+		return
+	}
+
+	var list map[string]*cacheObject
+	if err := json.Unmarshal(content, &list); err != nil {
+		return
+	}
+
+	for key, co := range list {
+		fc.set(key, co)
+	}
+}
+
+func (fc *fileCache) close() {
+	if dump, err := json.Marshal(fc.cache.list()); err == nil {
+		_ = ioutil.WriteFile(fc.home+cachePath, dump, 0644)
+	}
+}
+
+func (fc *fileCache) get(key string) (*cacheObject, bool) {
+	val, found := fc.cache.get(key)
```

Comment:
Maybe checking the timestamp before returning the value from the cache

Reply:
Yes. That's the idea now that I wrote this. That should be internal and not leak.